solana/solana-ibc/programs/solana-ibc/src/lib.rs

```diff
@@ -390,6 +392,33 @@ pub mod solana_ibc {
         .map_err(error::Error::TokenTransferError)
         .map_err(|err| error!((&err)))
     }
+
+    /// The body of this method is empty since it is called to
+    /// realloc the accounts only.  Anchor sets up the accounts
+    /// given in this call’s context before the body of the method is
+    /// executed.
+    pub fn realloc_accounts(
+        ctx: Context<ReallocAccounts>,
+        new_length: usize,
+    ) -> Result<()> {
+        let payer = &ctx.accounts.payer.to_account_info();
+        let account = &ctx.accounts.account.to_account_info();
+        let new_length = new_length.max(account.data_len());
+        let lamports = Rent::get()?
+            .minimum_balance(new_length)
+            .saturating_sub(account.lamports());
+        if lamports > 0 {
+            solana_program::program::invoke(
+                &system_instruction::transfer(
+                    &payer.key(),
+                    &account.key(),
+                    lamports,
+                ),
+                &[payer.clone(), account.clone()],
+            )?;
+        }
```

Comment:
As discussed, I believe we need to always take the additional lamports.

```suggestion
        let old_length = account.data_len();
        if new_length > old_length {
            let rent = Rent::get()?;
            let old_rent = rent.minimum_balance(old_length)
            let new_rent = rent.minimum_balance(new_length)
            solana_program::program::invoke(
                &system_instruction::transfer(
                    &payer.key(),
                    &account.key(),
                    new_rent - old_rent,
                ),
                &[payer.clone(), account.clone()],
            )?;
        }
```

Reply:
gotcha